### PR TITLE
Dismemberment moodlet now properly clears for regenerated limbs

### LINKED ISSUE #79812: Etherials don't lose the "I WAS USING THAT LIMB" mood debuff after regrowing a limb in a ressurection crystal
## Round ID:

 Multiple, pretty easy to test. Has happened to me in private testing for my crossbow pr.

## Reproduction:

Lose a limb to explosion as ethereal and go into crit and revive. Enjoy your permanent mood debuff.

### PR DESCRIPTION
**Pull request by wesoda25**
## About The Pull Request

That's a mouthful. Upon being resurrected by a crystal, dismembered ethereals now lose the dismemberment negative moodlet. Fixes: #79812

## Changelog
:cl:
fix: the dismemberment moodlet will now properly clear for ethereals who regrew a limb in their resurrection crystals
/:cl:

### CODE REVIEW

**Comment by Loafin34** [comment]
will this fix changeling regen limb moodlet?

**Comment by wesoda25** [comment]
Nope, but I can fix that too while I'm at it

**Review by MrMelbert — COMMENTED** [review]
This seems more like a general bug with the fact that `regenerate_limb` calls `try_attach_limb` with `special = TRUE`, meanwhile body parts don't clear the dismemberment moodlet if special is set

So like it should probably clear the moodlet regardless of special is set

**Comment by wesoda25** [comment]
Kay, ill try that and test to see if original intended behavior is changed.

**Comment by wesoda25** [comment]
Tested, both ethereal and changeling dismemberment properly clear now

File: code/modules/surgery/bodyparts/dismemberment.dm [hunk 1]
```diff
@@ -361,7 +361,8 @@
 		scar.victim = new_limb_owner
 		LAZYADD(new_limb_owner.all_scars, scar)
 
-	if(!special && new_limb_owner.mob_mood.has_mood_of_category("dismembered_[body_zone]"))
+	// clears dismemberment moodlet
```
Comment (Mothblocks) [inline code/modules/surgery/bodyparts/dismemberment.dm:364]:
```suggestion
```

Code is obvious, comments should explain why and not what

File: code/modules/surgery/bodyparts/dismemberment.dm [hunk 2]
```diff
@@ -361,7 +361,7 @@
 		scar.victim = new_limb_owner
 		LAZYADD(new_limb_owner.all_scars, scar)
 
-	if(!special && new_limb_owner.mob_mood.has_mood_of_category("dismembered_[body_zone]"))
+	if(new_limb_owner.mob_mood.has_mood_of_category("dismembered_[body_zone]"))
```
Comment (Mothblocks) [inline code/modules/surgery/bodyparts/dismemberment.dm:364]:
```suggestion
	if(new_limb_owner.mob_mood?.has_mood_of_category("dismembered_[body_zone]"))
```

**Pull request merged**